Traceback (most recent call last):
  File "/Users/fsultan/opt/anaconda3/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/fsultan/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/fsultan/opt/anaconda3/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/fsultan/opt/anaconda3/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Users/fsultan/opt/anaconda3/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/fsultan/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/Users/fsultan/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/fsultan/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
def binary_search_head(data, target):
    """ Recursive binary search algorithm """
    if len(data) == 0:
        return False
    else:
        mid = len(data) // 2
        if data[mid] == target:
            return True
        elif data[mid] > target:
            return binary_search_head(data[:mid], target)
        else:
            return binary_search_head(data[mid+1:], target)
        
assert binary_search_head([10, -20, 0, 5, 100], 0)   == True,  "Test case 1 failed"
assert binary_search_head([10, -20, 0, 5, 100], 10)  == True,  "Test case 2 failed"
assert binary_search_head([10, -20, 0, 5, 100], -1)  == False, "Test case 3 failed"
assert binary_search_head([10, -20, 0, 5, 100], -20) == True,  "Test case 4 failed"
assert binary_search_head([10, -20, 0, 5, 100], 101) == False, "Test case 5 failed"

print("All test cases passed!")
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mAssertionError[0m                            Traceback (most recent call last)
Input [0;32mIn [11][0m, in [0;36m<cell line: 15>[0;34m()[0m
[1;32m     12[0m             [38;5;28;01mreturn[39;00m binary_search_head(data[mid[38;5;241m+[39m[38;5;241m1[39m:], target)
[1;32m     14[0m [38;5;28;01massert[39;00m binary_search_head([[38;5;241m10[39m, [38;5;241m-[39m[38;5;241m20[39m, [38;5;241m0[39m, [38;5;241m5[39m, [38;5;241m100[39m], [38;5;241m0[39m)   [38;5;241m==[39m [38;5;28;01mTrue[39;00m,  [38;5;124m"[39m[38;5;124mTest case 1 failed[39m[38;5;124m"[39m
[0;32m---> 15[0m [38;5;28;01massert[39;00m binary_search_head([[38;5;241m10[39m, [38;5;241m-[39m[38;5;241m20[39m, [38;5;241m0[39m, [38;5;241m5[39m, [38;5;241m100[39m], [38;5;241m10[39m)  [38;5;241m==[39m [38;5;28;01mTrue[39;00m,  [38;5;124m"[39m[38;5;124mTest case 2 failed[39m[38;5;124m"[39m
[1;32m     16[0m [38;5;28;01massert[39;00m binary_search_head([[38;5;241m10[39m, [38;5;241m-[39m[38;5;241m20[39m, [38;5;241m0[39m, [38;5;241m5[39m, [38;5;241m100[39m], [38;5;241m-[39m[38;5;241m1[39m)  [38;5;241m==[39m [38;5;28;01mFalse[39;00m, [38;5;124m"[39m[38;5;124mTest case 3 failed[39m[38;5;124m"[39m
[1;32m     17[0m [38;5;28;01massert[39;00m binary_search_head([[38;5;241m10[39m, [38;5;241m-[39m[38;5;241m20[39m, [38;5;241m0[39m, [38;5;241m5[39m, [38;5;241m100[39m], [38;5;241m-[39m[38;5;241m20[39m) [38;5;241m==[39m [38;5;28;01mTrue[39;00m,  [38;5;124m"[39m[38;5;124mTest case 4 failed[39m[38;5;124m"[39m

[0;31mAssertionError[0m: Test case 2 failed
AssertionError: Test case 2 failed

